### ISSUE TITLE
Behavior Options

### ISSUE THREAD

**Issue by gossi** [almende/vis#530]
The network (maybe timeline, too) vis has some default behavior, when certain events happen, e.g. clicking a node selects all connected edges, selecting something fits the whole graph in the canvas (or at least in the hierarchical type).

I think these options are great and there is a good reason why they qualified as default. However, sometimes you may want to do something different or want to turn off the default behavior to implement your own. Sometimes the default behavior is completely wrong to something I want to do with my vis and "a feature I cannot deactive is a bug", there should be no hacks required to turn things off. For that I suggest the `behavior` options.

For the network I see two behaviors, that can be addressed. At least for tuning them on and off but can also provide alternatives (though that may require an investigation what's possible and if it is needed to require them here). Here are my two behaviors for the network vis:

1. `OnNodeSelect` options:
    a) `highlightEdges`:  Highlights all connected edges (default)
    b) `none`: Turns off this behavior

2. `OnSelect` (edges and nodes) options:
    a) `fitNetwork`: Fits the network in the current canvas by rescaling (current default)
    b) `none`: Turns off this behavior

More options can also be added but would be collected in one place.

**Comment by AlexDM0** [comment]
Hi,

OnSelect does not zoomExtend at all. On hierarchical layout there is a zoomExtend when altering the dataset because the hierarchical layout is reinitiated. There will be an option to disable this in the next release (the zoomExtend part). This part could also be improved. 

I like your suggestion for how to organise the options. We'll discuss it internally. 

Regards and happy new year!

Alex

**Comment by AlexDM0** [comment]
Hi,

Over the last year a lot of feature requests have been made. We have just introduced our new website which has a list of the requested features. We have placed this request on that list. 

The list can be found here:
http://visjs.org/featureRequests.html

An explaination of the new system can be found here:
http://visjs.org/blog.html#New\ website\ for\ vis\.js\!

I would like to stress that this does not mean we abandon this request. Discussion here will continue if needed on this feature but we will close it to keep our Github issue page more of a bug-todo list.

Future feature requests will still be made here and then added to the website by us.

Regards,

Alex
